catalog/app/components/JsonDisplay/JsonDisplay.js

```diff
@@ -157,11 +157,19 @@ function CompoundEntry({
   defaultExpanded = false,
   showKeysWhenCollapsed,
   classes,
+  onToggle,
 }) {
   const braces = Array.isArray(value) ? '[]' : '{}'
   const entries = React.useMemo(() => Object.entries(value), [value])
   const [stateExpanded, setExpanded] = React.useState(!!defaultExpanded)
-  const toggle = React.useCallback(() => setExpanded((e) => !e), [])
+  const toggle = React.useCallback(() => {
+    const newStateExpanded = !stateExpanded
+    if (onToggle) {
+      onToggle(newStateExpanded)
```

Comment:
i don't quite get how this works: if you provide the onToggle prop, how the state is changed?

Reply:
if you plan to rely on `defaultExpanded`, this won't work afaict, bc it's only used once to initialize the state, and all the updates to this prop are ignored

Reply:
It works accidentally :)
But I realized my mistake, thank you

I planned to use `onToggle` as a callback without changing the state, so I'll return to that decision